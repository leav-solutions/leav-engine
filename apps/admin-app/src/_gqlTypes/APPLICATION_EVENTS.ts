// Copyright LEAV Solutions 2017
// This file is released under LGPL V3
// License text available at https://www.gnu.org/licenses/lgpl-3.0.txt
/* tslint:disable */
/* eslint-disable */
// @generated
// This file was automatically generated and should not be edited.

import {
    ApplicationEventFiltersInput,
    ApplicationEventTypes,
    ApplicationType,
    ApplicationInstallStatus
} from './globalTypes';

// ====================================================
// GraphQL subscription operation: APPLICATION_EVENTS
// ====================================================

export interface APPLICATION_EVENTS_applicationEvent_application_icon_whoAmI_library {
    id: string;
    label: SystemTranslation | null;
}

export interface APPLICATION_EVENTS_applicationEvent_application_icon_whoAmI_preview {
    tiny: string | null;
    small: string | null;
    medium: string | null;
    big: string | null;
    huge: string | null;
    pdf: string | null;
}

export interface APPLICATION_EVENTS_applicationEvent_application_icon_whoAmI {
    id: string;
    library: APPLICATION_EVENTS_applicationEvent_application_icon_whoAmI_library;
    label: string | null;
    color: string | null;
    preview: APPLICATION_EVENTS_applicationEvent_application_icon_whoAmI_preview | null;
}

export interface APPLICATION_EVENTS_applicationEvent_application_icon {
    whoAmI: APPLICATION_EVENTS_applicationEvent_application_icon_whoAmI;
}

export interface APPLICATION_EVENTS_applicationEvent_application_libraries {
    id: string;
}

export interface APPLICATION_EVENTS_applicationEvent_application_trees {
    id: string;
}

export interface APPLICATION_EVENTS_applicationEvent_application_permissions {
    access_application: boolean;
    admin_application: boolean;
}

export interface APPLICATION_EVENTS_applicationEvent_application_install {
    status: ApplicationInstallStatus;
    lastCallResult: string | null;
}

export interface APPLICATION_EVENTS_applicationEvent_application {
    id: string;
    label: SystemTranslation;
    type: ApplicationType;
    description: SystemTranslation | null;
    endpoint: string | null;
    url: string | null;
    color: string | null;
    icon: APPLICATION_EVENTS_applicationEvent_application_icon | null;
    module: string | null;
    libraries: APPLICATION_EVENTS_applicationEvent_application_libraries[] | null;
    trees: APPLICATION_EVENTS_applicationEvent_application_trees[] | null;
    permissions: APPLICATION_EVENTS_applicationEvent_application_permissions;
    install: APPLICATION_EVENTS_applicationEvent_application_install | null;
    settings: JSONObject | null;
}

export interface APPLICATION_EVENTS_applicationEvent {
    type: ApplicationEventTypes;
    application: APPLICATION_EVENTS_applicationEvent_application;
}

export interface APPLICATION_EVENTS {
    applicationEvent: APPLICATION_EVENTS_applicationEvent;
}

export interface APPLICATION_EVENTSVariables {
    filters?: ApplicationEventFiltersInput | null;
}
