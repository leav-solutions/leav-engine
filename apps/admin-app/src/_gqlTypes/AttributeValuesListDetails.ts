// Copyright LEAV Solutions 2017
// This file is released under LGPL V3
// License text available at https://www.gnu.org/licenses/lgpl-3.0.txt
/* tslint:disable */
/* eslint-disable */
// @generated
// This file was automatically generated and should not be edited.

// ====================================================
// GraphQL fragment: AttributeValuesListDetails
// ====================================================

export interface AttributeValuesListDetails_StandardAttribute_values_list_StandardStringValuesListConf {
    enable: boolean;
    allowFreeEntry: boolean | null;
    values: string[] | null;
}

export interface AttributeValuesListDetails_StandardAttribute_values_list_StandardDateRangeValuesListConf_dateRangeValues {
    from: string | null;
    to: string | null;
}

export interface AttributeValuesListDetails_StandardAttribute_values_list_StandardDateRangeValuesListConf {
    enable: boolean;
    allowFreeEntry: boolean | null;
    dateRangeValues:
        | AttributeValuesListDetails_StandardAttribute_values_list_StandardDateRangeValuesListConf_dateRangeValues[]
        | null;
}

export type AttributeValuesListDetails_StandardAttribute_values_list =
    | AttributeValuesListDetails_StandardAttribute_values_list_StandardStringValuesListConf
    | AttributeValuesListDetails_StandardAttribute_values_list_StandardDateRangeValuesListConf;

export interface AttributeValuesListDetails_StandardAttribute {
    values_list: AttributeValuesListDetails_StandardAttribute_values_list | null;
}

export interface AttributeValuesListDetails_LinkAttribute_values_list_linkValues_whoAmI_library {
    id: string;
    label: SystemTranslation | null;
}

export interface AttributeValuesListDetails_LinkAttribute_values_list_linkValues_whoAmI_preview {
    small: string | null;
    medium: string | null;
    pdf: string | null;
    big: string | null;
}

export interface AttributeValuesListDetails_LinkAttribute_values_list_linkValues_whoAmI {
    id: string;
    library: AttributeValuesListDetails_LinkAttribute_values_list_linkValues_whoAmI_library;
    label: string | null;
    color: string | null;
    preview: AttributeValuesListDetails_LinkAttribute_values_list_linkValues_whoAmI_preview | null;
}

export interface AttributeValuesListDetails_LinkAttribute_values_list_linkValues {
    whoAmI: AttributeValuesListDetails_LinkAttribute_values_list_linkValues_whoAmI;
}

export interface AttributeValuesListDetails_LinkAttribute_values_list {
    enable: boolean;
    allowFreeEntry: boolean | null;
    linkValues: AttributeValuesListDetails_LinkAttribute_values_list_linkValues[] | null;
}

export interface AttributeValuesListDetails_LinkAttribute {
    values_list: AttributeValuesListDetails_LinkAttribute_values_list | null;
}

export interface AttributeValuesListDetails_TreeAttribute_values_list_treeValues_record_whoAmI_library {
    id: string;
    label: SystemTranslation | null;
}

export interface AttributeValuesListDetails_TreeAttribute_values_list_treeValues_record_whoAmI_preview {
    small: string | null;
    medium: string | null;
    pdf: string | null;
    big: string | null;
}

export interface AttributeValuesListDetails_TreeAttribute_values_list_treeValues_record_whoAmI {
    id: string;
    library: AttributeValuesListDetails_TreeAttribute_values_list_treeValues_record_whoAmI_library;
    label: string | null;
    color: string | null;
    preview: AttributeValuesListDetails_TreeAttribute_values_list_treeValues_record_whoAmI_preview | null;
}

export interface AttributeValuesListDetails_TreeAttribute_values_list_treeValues_record {
    whoAmI: AttributeValuesListDetails_TreeAttribute_values_list_treeValues_record_whoAmI;
}

export interface AttributeValuesListDetails_TreeAttribute_values_list_treeValues_ancestors_record_whoAmI_library {
    id: string;
    label: SystemTranslation | null;
}

export interface AttributeValuesListDetails_TreeAttribute_values_list_treeValues_ancestors_record_whoAmI_preview {
    small: string | null;
    medium: string | null;
    pdf: string | null;
    big: string | null;
}

export interface AttributeValuesListDetails_TreeAttribute_values_list_treeValues_ancestors_record_whoAmI {
    id: string;
    library: AttributeValuesListDetails_TreeAttribute_values_list_treeValues_ancestors_record_whoAmI_library;
    label: string | null;
    color: string | null;
    preview: AttributeValuesListDetails_TreeAttribute_values_list_treeValues_ancestors_record_whoAmI_preview | null;
}

export interface AttributeValuesListDetails_TreeAttribute_values_list_treeValues_ancestors_record {
    whoAmI: AttributeValuesListDetails_TreeAttribute_values_list_treeValues_ancestors_record_whoAmI;
}

export interface AttributeValuesListDetails_TreeAttribute_values_list_treeValues_ancestors {
    record: AttributeValuesListDetails_TreeAttribute_values_list_treeValues_ancestors_record;
}

export interface AttributeValuesListDetails_TreeAttribute_values_list_treeValues {
    record: AttributeValuesListDetails_TreeAttribute_values_list_treeValues_record;
    ancestors: AttributeValuesListDetails_TreeAttribute_values_list_treeValues_ancestors[] | null;
}

export interface AttributeValuesListDetails_TreeAttribute_values_list {
    enable: boolean;
    allowFreeEntry: boolean | null;
    treeValues: AttributeValuesListDetails_TreeAttribute_values_list_treeValues[] | null;
}

export interface AttributeValuesListDetails_TreeAttribute {
    values_list: AttributeValuesListDetails_TreeAttribute_values_list | null;
}

export type AttributeValuesListDetails =
    | AttributeValuesListDetails_StandardAttribute
    | AttributeValuesListDetails_LinkAttribute
    | AttributeValuesListDetails_TreeAttribute;
