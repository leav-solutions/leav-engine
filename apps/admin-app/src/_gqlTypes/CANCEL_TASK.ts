// Copyright LEAV Solutions 2017
// This file is released under LGPL V3
// License text available at https://www.gnu.org/licenses/lgpl-3.0.txt
/* tslint:disable */
/* eslint-disable */
// @generated
// This file was automatically generated and should not be edited.

// ====================================================
// GraphQL mutation operation: CANCEL_TASK
// ====================================================

export interface CANCEL_TASK {
    cancelTask: boolean;
}

export interface CANCEL_TASKVariables {
    taskId: string;
}
