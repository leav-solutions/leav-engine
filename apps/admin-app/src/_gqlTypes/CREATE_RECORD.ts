// Copyright LEAV Solutions 2017
// This file is released under LGPL V3
// License text available at https://www.gnu.org/licenses/lgpl-3.0.txt
/* tslint:disable */
/* eslint-disable */
// @generated
// This file was automatically generated and should not be edited.

// ====================================================
// GraphQL mutation operation: CREATE_RECORD
// ====================================================

export interface CREATE_RECORD_createRecord_whoAmI_library {
    id: string;
    label: SystemTranslation | null;
}

export interface CREATE_RECORD_createRecord_whoAmI_preview {
    tiny: string | null;
    small: string | null;
    medium: string | null;
    big: string | null;
    huge: string | null;
    pdf: string | null;
}

export interface CREATE_RECORD_createRecord_whoAmI {
    id: string;
    library: CREATE_RECORD_createRecord_whoAmI_library;
    label: string | null;
    color: string | null;
    preview: CREATE_RECORD_createRecord_whoAmI_preview | null;
}

export interface CREATE_RECORD_createRecord {
    id: string;
    whoAmI: CREATE_RECORD_createRecord_whoAmI;
}

export interface CREATE_RECORD {
    createRecord: CREATE_RECORD_createRecord;
}

export interface CREATE_RECORDVariables {
    library: string;
}
