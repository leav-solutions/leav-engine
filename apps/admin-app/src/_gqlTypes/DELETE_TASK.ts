// Copyright LEAV Solutions 2017
// This file is released under LGPL V3
// License text available at https://www.gnu.org/licenses/lgpl-3.0.txt
/* tslint:disable */
/* eslint-disable */
// @generated
// This file was automatically generated and should not be edited.

// ====================================================
// GraphQL mutation operation: DELETE_TASK
// ====================================================

export interface DELETE_TASK_deleteTask {
    id: string;
}

export interface DELETE_TASK {
    deleteTask: DELETE_TASK_deleteTask;
}

export interface DELETE_TASKVariables {
    taskId: string;
    archive: boolean;
}
