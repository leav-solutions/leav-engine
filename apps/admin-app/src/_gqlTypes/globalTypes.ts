// Copyright LEAV Solutions 2017
// This file is released under LGPL V3
// License text available at https://www.gnu.org/licenses/lgpl-3.0.txt
/* tslint:disable */
/* eslint-disable */
// @generated
// This file was automatically generated and should not be edited.

//==============================================================
// START Enums and Input Objects
//==============================================================

export enum ActionIOTypes {
    boolean = 'boolean',
    number = 'number',
    object = 'object',
    string = 'string'
}

export enum ApplicationInstallStatus {
    ERROR = 'ERROR',
    NONE = 'NONE',
    RUNNING = 'RUNNING',
    SUCCESS = 'SUCCESS'
}

export enum ApplicationSortableFields {
    endpoint = 'endpoint',
    id = 'id',
    module = 'module',
    system = 'system',
    type = 'type'
}

export enum ApplicationType {
    external = 'external',
    internal = 'internal'
}

export enum AttributeFormat {
    boolean = 'boolean',
    date = 'date',
    date_range = 'date_range',
    encrypted = 'encrypted',
    extended = 'extended',
    numeric = 'numeric',
    text = 'text'
}

export enum AttributeType {
    advanced = 'advanced',
    advanced_link = 'advanced_link',
    simple = 'simple',
    simple_link = 'simple_link',
    tree = 'tree'
}

export enum AvailableLanguage {
    en = 'en',
    fr = 'fr'
}

export enum FormElementTypes {
    field = 'field',
    layout = 'layout'
}

export enum IOTypes {
    boolean = 'boolean',
    number = 'number',
    object = 'object',
    string = 'string'
}

export enum LibraryBehavior {
    directories = 'directories',
    files = 'files',
    standard = 'standard'
}

export enum PermissionTypes {
    admin = 'admin',
    application = 'application',
    attribute = 'attribute',
    library = 'library',
    record = 'record',
    record_attribute = 'record_attribute',
    tree = 'tree',
    tree_library = 'tree_library',
    tree_node = 'tree_node'
}

export enum PermissionsActions {
    access_application = 'access_application',
    access_attribute = 'access_attribute',
    access_library = 'access_library',
    access_record = 'access_record',
    access_tree = 'access_tree',
    admin_access_applications = 'admin_access_applications',
    admin_access_attributes = 'admin_access_attributes',
    admin_access_libraries = 'admin_access_libraries',
    admin_access_permissions = 'admin_access_permissions',
    admin_access_trees = 'admin_access_trees',
    admin_application = 'admin_application',
    admin_create_application = 'admin_create_application',
    admin_create_attribute = 'admin_create_attribute',
    admin_create_library = 'admin_create_library',
    admin_create_tree = 'admin_create_tree',
    admin_delete_application = 'admin_delete_application',
    admin_delete_attribute = 'admin_delete_attribute',
    admin_delete_library = 'admin_delete_library',
    admin_delete_tree = 'admin_delete_tree',
    admin_edit_application = 'admin_edit_application',
    admin_edit_attribute = 'admin_edit_attribute',
    admin_edit_library = 'admin_edit_library',
    admin_edit_permission = 'admin_edit_permission',
    admin_edit_tree = 'admin_edit_tree',
    admin_library = 'admin_library',
    admin_manage_global_preferences = 'admin_manage_global_preferences',
    create_record = 'create_record',
    delete_record = 'delete_record',
    detach = 'detach',
    edit_children = 'edit_children',
    edit_record = 'edit_record',
    edit_value = 'edit_value'
}

export enum PermissionsRelation {
    and = 'and',
    or = 'or'
}

export enum SortOrder {
    asc = 'asc',
    desc = 'desc'
}

export enum TreeBehavior {
    files = 'files',
    standard = 'standard'
}

export enum ValueVersionMode {
    simple = 'simple',
    smart = 'smart'
}

export interface ActionConfigurationInput {
    id: string;
    params?: ActionConfigurationParamInput[] | null;
}

export interface ActionConfigurationParamInput {
    name: string;
    value: string;
}

export interface ActionsListConfigurationInput {
    saveValue?: ActionConfigurationInput[] | null;
    getValue?: ActionConfigurationInput[] | null;
    deleteValue?: ActionConfigurationInput[] | null;
}

export interface ApplicationIconInput {
    libraryId: string;
    recordId: string;
}

export interface ApplicationInput {
    id: string;
    label?: SystemTranslation | null;
    type?: ApplicationType | null;
    description?: SystemTranslation | null;
    libraries?: string[] | null;
    trees?: string[] | null;
    color?: string | null;
    icon?: ApplicationIconInput | null;
    module?: string | null;
    endpoint?: string | null;
    settings?: JSONObject | null;
}

export interface ApplicationsFiltersInput {
    id?: string | null;
    label?: string | null;
    type?: (ApplicationType | null)[] | null;
    system?: boolean | null;
    endpoint?: string | null;
    module?: string | null;
}

export interface AttributeInput {
    id: string;
    type?: AttributeType | null;
    format?: AttributeFormat | null;
    label?: SystemTranslation | null;
    description?: SystemTranslation | null;
    linked_library?: string | null;
    linked_tree?: string | null;
    embedded_fields?: (EmbeddedAttributeInput | null)[] | null;
    actions_list?: ActionsListConfigurationInput | null;
    permissions_conf?: Treepermissions_confInput | null;
    multiple_values?: boolean | null;
    versions_conf?: ValuesVersionsConfInput | null;
    metadata_fields?: string[] | null;
    values_list?: ValuesListConfInput | null;
    reverse_link?: string | null;
}

export interface EmbeddedAttributeInput {
    id: string;
    format?: AttributeFormat | null;
    label?: SystemTranslation | null;
    description?: SystemTranslation | null;
    validation_regex?: string | null;
    embedded_fields?: (EmbeddedAttributeInput | null)[] | null;
}

export interface FormDependencyValueInput {
    attribute: string;
    value: TreeElementInput;
}

export interface FormElementInput {
    id: string;
    containerId: string;
    order: number;
    uiElementType: string;
    type: FormElementTypes;
    settings: FormElementSettingsInput[];
}

export interface FormElementSettingsInput {
    key: string;
    value: Any;
}

export interface FormElementsByDepsInput {
    dependencyValue?: FormDependencyValueInput | null;
    elements: FormElementInput[];
}

export interface FormInput {
    id: string;
    library: string;
    label?: SystemTranslation | null;
    dependencyAttributes?: string[] | null;
    elements?: FormElementsByDepsInput[] | null;
}

export interface LibraryInput {
    id: string;
    label?: SystemTranslation | null;
    attributes?: string[] | null;
    fullTextAttributes?: string[] | null;
    behavior?: LibraryBehavior | null;
    permissions_conf?: Treepermissions_confInput | null;
    recordIdentityConf?: RecordIdentityConfInput | null;
    defaultView?: string | null;
}

export interface Pagination {
    limit: number;
    offset: number;
}

export interface PermissionActionInput {
    name: PermissionsActions;
    allowed?: boolean | null;
}

/**
 * If users group is not specified, permission will be saved at root level.
 * If saving a tree-based permission (record or attribute) and tree target's id is not specified,
 * permission will be saved at root level for any element of the tree.
 */
export interface PermissionInput {
    type: PermissionTypes;
    applyTo?: string | null;
    usersGroup?: string | null;
    actions: PermissionActionInput[];
    permissionTreeTarget?: PermissionsTreeTargetInput | null;
}

/**
 * Element on which we want to retrieve record or attribute permission. Record ID is mandatory,
 * attributeId is only required for attribute permission
 * libraryId and recordId are mandatory for tree node permission
 */
export interface PermissionTarget {
    attributeId?: string | null;
    recordId?: string | null;
    libraryId?: string | null;
    nodeId?: string | null;
}

/**
 * If id and library are not specified, permission will apply to tree root
 */
export interface PermissionsTreeTargetInput {
    tree: string;
    nodeId?: string | null;
}

export interface RecordIdentityConfInput {
    label?: string | null;
    color?: string | null;
    preview?: string | null;
    treeColorPreview?: string | null;
}

export interface SortApplications {
    field: ApplicationSortableFields;
    order?: SortOrder | null;
}

export interface TreeElementInput {
    id: string;
    library: string;
}

export interface TreeInput {
    id: string;
    libraries?: TreeLibraryInput[] | null;
    behavior?: TreeBehavior | null;
    label?: SystemTranslation | null;
    permissions_conf?: TreeNodePermissionsConfInput[] | null;
}

export interface TreeLibraryInput {
    library: string;
    settings: TreeLibrarySettingsInput;
}

export interface TreeLibrarySettingsInput {
    allowMultiplePositions: boolean;
    allowedChildren: string[];
    allowedAtRoot: boolean;
}

export interface TreeNodePermissionsConfInput {
    libraryId: string;
    permissionsConf: Treepermissions_confInput;
}

export interface Treepermissions_confInput {
    permissionTreeAttributes: string[];
    relation: PermissionsRelation;
}

export interface ValueBatchInput {
    attribute?: string | null;
    id_value?: string | null;
    value?: string | null;
    metadata?: (ValueMetadataInput | null)[] | null;
}

export interface ValueInput {
    id_value?: string | null;
    value?: string | null;
    metadata?: (ValueMetadataInput | null)[] | null;
    version?: (ValueVersionInput | null)[] | null;
}

export interface ValueMetadataInput {
    name: string;
    value?: string | null;
}

export interface ValueVersionInput {
    name: string;
    value: string;
}

export interface ValuesListConfInput {
    enable: boolean;
    allowFreeEntry?: boolean | null;
    values?: string[] | null;
}

export interface ValuesVersionsConfInput {
    versionable: boolean;
    mode?: ValueVersionMode | null;
    trees?: string[] | null;
}

//==============================================================
// END Enums and Input Objects
//==============================================================
