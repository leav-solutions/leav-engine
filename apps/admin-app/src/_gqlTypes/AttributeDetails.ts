// Copyright LEAV Solutions 2017
// This file is released under LGPL V3
// License text available at https://www.gnu.org/licenses/lgpl-3.0.txt
/* tslint:disable */
/* eslint-disable */
// @generated
// This file was automatically generated and should not be edited.

import {AttributeType, AttributeFormat, PermissionsRelation, ValueVersionMode} from './globalTypes';

// ====================================================
// GraphQL fragment: AttributeDetails
// ====================================================

export interface AttributeDetails_StandardAttribute_metadata_fields {
    id: string;
    label: SystemTranslation | null;
    type: AttributeType;
    format: AttributeFormat | null;
}

export interface AttributeDetails_StandardAttribute_permissions_conf_permissionTreeAttributes_StandardAttribute {
    id: string;
    label: SystemTranslation | null;
}

export interface AttributeDetails_StandardAttribute_permissions_conf_permissionTreeAttributes_TreeAttribute_linked_tree {
    id: string;
}

export interface AttributeDetails_StandardAttribute_permissions_conf_permissionTreeAttributes_TreeAttribute {
    id: string;
    label: SystemTranslation | null;
    linked_tree: AttributeDetails_StandardAttribute_permissions_conf_permissionTreeAttributes_TreeAttribute_linked_tree | null;
}

export type AttributeDetails_StandardAttribute_permissions_conf_permissionTreeAttributes =
    | AttributeDetails_StandardAttribute_permissions_conf_permissionTreeAttributes_StandardAttribute
    | AttributeDetails_StandardAttribute_permissions_conf_permissionTreeAttributes_TreeAttribute;

export interface AttributeDetails_StandardAttribute_permissions_conf {
    permissionTreeAttributes: AttributeDetails_StandardAttribute_permissions_conf_permissionTreeAttributes[];
    relation: PermissionsRelation;
}

export interface AttributeDetails_StandardAttribute_versions_conf {
    versionable: boolean;
    mode: ValueVersionMode | null;
    trees: string[] | null;
}

export interface AttributeDetails_StandardAttribute_libraries {
    id: string;
    label: SystemTranslation | null;
}

export interface AttributeDetails_StandardAttribute {
    id: string;
    type: AttributeType;
    format: AttributeFormat | null;
    system: boolean;
    readonly: boolean;
    label: SystemTranslation | null;
    description: SystemTranslation | null;
    multiple_values: boolean;
    metadata_fields: AttributeDetails_StandardAttribute_metadata_fields[] | null;
    permissions_conf: AttributeDetails_StandardAttribute_permissions_conf | null;
    versions_conf: AttributeDetails_StandardAttribute_versions_conf | null;
    libraries: AttributeDetails_StandardAttribute_libraries[] | null;
}

export interface AttributeDetails_LinkAttribute_metadata_fields {
    id: string;
    label: SystemTranslation | null;
    type: AttributeType;
    format: AttributeFormat | null;
}

export interface AttributeDetails_LinkAttribute_permissions_conf_permissionTreeAttributes_StandardAttribute {
    id: string;
    label: SystemTranslation | null;
}

export interface AttributeDetails_LinkAttribute_permissions_conf_permissionTreeAttributes_TreeAttribute_linked_tree {
    id: string;
}

export interface AttributeDetails_LinkAttribute_permissions_conf_permissionTreeAttributes_TreeAttribute {
    id: string;
    label: SystemTranslation | null;
    linked_tree: AttributeDetails_LinkAttribute_permissions_conf_permissionTreeAttributes_TreeAttribute_linked_tree | null;
}

export type AttributeDetails_LinkAttribute_permissions_conf_permissionTreeAttributes =
    | AttributeDetails_LinkAttribute_permissions_conf_permissionTreeAttributes_StandardAttribute
    | AttributeDetails_LinkAttribute_permissions_conf_permissionTreeAttributes_TreeAttribute;

export interface AttributeDetails_LinkAttribute_permissions_conf {
    permissionTreeAttributes: AttributeDetails_LinkAttribute_permissions_conf_permissionTreeAttributes[];
    relation: PermissionsRelation;
}

export interface AttributeDetails_LinkAttribute_versions_conf {
    versionable: boolean;
    mode: ValueVersionMode | null;
    trees: string[] | null;
}

export interface AttributeDetails_LinkAttribute_libraries {
    id: string;
    label: SystemTranslation | null;
}

export interface AttributeDetails_LinkAttribute_linked_library {
    id: string;
}

export interface AttributeDetails_LinkAttribute {
    id: string;
    type: AttributeType;
    format: AttributeFormat | null;
    system: boolean;
    readonly: boolean;
    label: SystemTranslation | null;
    description: SystemTranslation | null;
    multiple_values: boolean;
    metadata_fields: AttributeDetails_LinkAttribute_metadata_fields[] | null;
    permissions_conf: AttributeDetails_LinkAttribute_permissions_conf | null;
    versions_conf: AttributeDetails_LinkAttribute_versions_conf | null;
    libraries: AttributeDetails_LinkAttribute_libraries[] | null;
    linked_library: AttributeDetails_LinkAttribute_linked_library | null;
    reverse_link: string | null;
}

export interface AttributeDetails_TreeAttribute_metadata_fields {
    id: string;
    label: SystemTranslation | null;
    type: AttributeType;
    format: AttributeFormat | null;
}

export interface AttributeDetails_TreeAttribute_permissions_conf_permissionTreeAttributes_StandardAttribute {
    id: string;
    label: SystemTranslation | null;
}

export interface AttributeDetails_TreeAttribute_permissions_conf_permissionTreeAttributes_TreeAttribute_linked_tree {
    id: string;
}

export interface AttributeDetails_TreeAttribute_permissions_conf_permissionTreeAttributes_TreeAttribute {
    id: string;
    label: SystemTranslation | null;
    linked_tree: AttributeDetails_TreeAttribute_permissions_conf_permissionTreeAttributes_TreeAttribute_linked_tree | null;
}

export type AttributeDetails_TreeAttribute_permissions_conf_permissionTreeAttributes =
    | AttributeDetails_TreeAttribute_permissions_conf_permissionTreeAttributes_StandardAttribute
    | AttributeDetails_TreeAttribute_permissions_conf_permissionTreeAttributes_TreeAttribute;

export interface AttributeDetails_TreeAttribute_permissions_conf {
    permissionTreeAttributes: AttributeDetails_TreeAttribute_permissions_conf_permissionTreeAttributes[];
    relation: PermissionsRelation;
}

export interface AttributeDetails_TreeAttribute_versions_conf {
    versionable: boolean;
    mode: ValueVersionMode | null;
    trees: string[] | null;
}

export interface AttributeDetails_TreeAttribute_libraries {
    id: string;
    label: SystemTranslation | null;
}

export interface AttributeDetails_TreeAttribute_linked_tree {
    id: string;
}

export interface AttributeDetails_TreeAttribute {
    id: string;
    type: AttributeType;
    format: AttributeFormat | null;
    system: boolean;
    readonly: boolean;
    label: SystemTranslation | null;
    description: SystemTranslation | null;
    multiple_values: boolean;
    metadata_fields: AttributeDetails_TreeAttribute_metadata_fields[] | null;
    permissions_conf: AttributeDetails_TreeAttribute_permissions_conf | null;
    versions_conf: AttributeDetails_TreeAttribute_versions_conf | null;
    libraries: AttributeDetails_TreeAttribute_libraries[] | null;
    linked_tree: AttributeDetails_TreeAttribute_linked_tree | null;
}

export type AttributeDetails =
    | AttributeDetails_StandardAttribute
    | AttributeDetails_LinkAttribute
    | AttributeDetails_TreeAttribute;
