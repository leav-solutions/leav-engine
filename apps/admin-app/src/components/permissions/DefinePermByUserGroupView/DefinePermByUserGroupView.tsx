// Copyright LEAV Solutions 2017
// This file is released under LGPL V3
// License text available at https://www.gnu.org/licenses/lgpl-3.0.txt
import React, {useState} from 'react';
import {ITreeNodeData} from '_types/trees';
import {getTreeNodeKey} from '../../../utils/utils';
import {PermissionTypes} from '../../../_gqlTypes/globalTypes';
import ColumnsDisplay from '../../shared/ColumnsDisplay';
import DefinePermissionsViewLoadTree from '../DefinePermissionsViewLoadTree';
import EditPermissions from '../EditPermissions';

interface IDefinePermByUserGroupViewProps {
    applyTo?: string;
    readOnly?: boolean;
    type: PermissionTypes;
}

function DefinePermByUserGroupView({applyTo, readOnly, type}: IDefinePermByUserGroupViewProps): JSX.Element {
    const usersGroupsTreeId = 'users_groups';

    const [selectedGroupNode, setSelectedGroupNode] = useState<ITreeNodeData | null>(null);

    const _selectGroupNode = (nodeData: ITreeNodeData) =>
        setSelectedGroupNode(getTreeNodeKey(nodeData) !== getTreeNodeKey(selectedGroupNode) ? nodeData : null);

    const cols = [
        <DefinePermissionsViewLoadTree
            key="users_groups"
            treeId={usersGroupsTreeId}
            onClick={_selectGroupNode}
            selectedNode={selectedGroupNode}
        />
    ];

    if (selectedGroupNode) {
        cols.push(
            <EditPermissions
                permParams={{
                    type,
                    applyTo,
                    usersGroup: selectedGroupNode.node.id !== 'root' ? selectedGroupNode.node.id : null
                }}
                readOnly={readOnly}
            />
        );
    }

    return <ColumnsDisplay columnsNumber={2} columnsContent={cols} />;
}

export default DefinePermByUserGroupView;
