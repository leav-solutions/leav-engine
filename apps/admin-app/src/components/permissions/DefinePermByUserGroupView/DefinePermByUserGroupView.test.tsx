// Copyright LEAV Solutions 2017
// This file is released under LGPL V3
// License text available at https://www.gnu.org/licenses/lgpl-3.0.txt
import {MockedProvider} from '@apollo/client/testing';
import {render} from 'enzyme';
import React from 'react';
import {PermissionTypes} from '../../../_gqlTypes/globalTypes';
import DefinePermByUserGroupView from './DefinePermByUserGroupView';

jest.mock('../../../hooks/useLang');

describe('DefineAttrPermissionsView', () => {
    test('Snapshot test', async () => {
        const comp = render(
            <MockedProvider>
                <DefinePermByUserGroupView type={PermissionTypes.attribute} applyTo="test_attr" readOnly={false} />
            </MockedProvider>
        );

        expect(comp).toMatchSnapshot();
    });
});
