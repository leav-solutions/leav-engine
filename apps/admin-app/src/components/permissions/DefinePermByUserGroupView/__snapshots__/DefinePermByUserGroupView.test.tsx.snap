// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DefineAttrPermissionsView Snapshot test 1`] = `
initialize {
  "0": Node {
    "attribs": Object {
      "class": "sc-bdVaJa goaYMK",
    },
    "children": Array [
      Node {
        "attribs": Object {
          "class": "sc-bwzfXH hfCnvE",
        },
        "children": Array [
          Node {
            "attribs": Object {
              "class": "ui active text centered inline loader",
            },
            "children": Array [
              Node {
                "data": "admin.loading",
                "next": null,
                "parent": [Circular],
                "prev": null,
                "type": "text",
              },
            ],
            "name": "div",
            "namespace": "http://www.w3.org/1999/xhtml",
            "next": null,
            "parent": [Circular],
            "prev": null,
            "type": "tag",
            "x-attribsNamespace": Object {
              "class": undefined,
            },
            "x-attribsPrefix": Object {
              "class": undefined,
            },
          },
        ],
        "name": "div",
        "namespace": "http://www.w3.org/1999/xhtml",
        "next": null,
        "parent": [Circular],
        "prev": null,
        "type": "tag",
        "x-attribsNamespace": Object {
          "class": undefined,
        },
        "x-attribsPrefix": Object {
          "class": undefined,
        },
      },
    ],
    "name": "div",
    "namespace": "http://www.w3.org/1999/xhtml",
    "next": null,
    "parent": Node {
      "children": Array [
        [Circular],
      ],
      "name": "root",
      "next": null,
      "parent": null,
      "prev": null,
      "type": "root",
    },
    "prev": null,
    "type": "tag",
    "x-attribsNamespace": Object {
      "class": undefined,
    },
    "x-attribsPrefix": Object {
      "class": undefined,
    },
  },
  "_root": [Circular],
  "length": 1,
  "options": Object {
    "decodeEntities": true,
    "xml": false,
  },
}
`;
